corehq/apps/geospatial/tasks.py

```diff
@@ -14,3 +28,42 @@ def geo_cases_reassignment_update_owners(domain, case_owner_updates_dict, task_k
     finally:
         celery_task_tracker = CeleryTaskTracker(task_key)
         celery_task_tracker.mark_completed()
+
+
+@task(queue='geospatial_queue', ignore_result=True)
+def index_es_docs_with_location_props(domain):
+    celery_task_tracker = get_celery_task_tracker(domain, INDEX_ES_TASK_HELPER_BASE_KEY)
+    if celery_task_tracker.is_active():
+        return
+
+    geo_case_prop = get_geo_case_property(domain)
+    query = case_query_for_missing_geopoint_val(domain, geo_case_prop)
+    doc_count = query.count()
+    if doc_count > MAX_GEOSPATIAL_INDEX_DOC_LIMIT:
+        celery_task_tracker.mark_as_error(error_slug='TOO_MANY_CASES')
+        return
+
+    celery_task_tracker.mark_requested()
+    batch_count = get_batch_count(doc_count, DEFAULT_QUERY_LIMIT)
+    try:
+        for i in range(batch_count):
+            docs_left = doc_count - (DEFAULT_QUERY_LIMIT * i)
+            limit = min(DEFAULT_QUERY_LIMIT, docs_left)
+            process_batch(
+                domain,
+                geo_case_prop,
+                case_type=None,
+                query_limit=limit,
+                chunk_size=DEFAULT_CHUNK_SIZE,
+                offset=i * DEFAULT_QUERY_LIMIT,
+            )
+            celery_task_tracker.update_progress(current=i + 1, total=batch_count)
+    except Exception as e:
+        celery_task_tracker.mark_as_error(error_slug='CELERY')
+        notify_exception(
+            None,
+            'Something went wrong with index_es_docs_with_location_props()',
```

Comment:
nit: Do we need the function name in the message? Isn't a stack trace available with the notification in sentry?

Reply:
This is in the stack trace, but I specifically called out the function in an attempt to make searching for it easier in Sentry. Having a look there again though, I see we can simply use a transaction filter to search for this function.

Given the above, I'll refactor the description a bit.

Reply:
Addressed in [901cbc1](https://github.com/dimagi/commcare-hq/pull/35126/commits/901cbc15ed1fa24f870e01de0514e96c5b757ea6). I have also added the domain to the exception logging for easier filtering.